contrib/scripts.py

```diff
@@ -9,15 +9,16 @@
 from typing import TYPE_CHECKING
 
 import requests
-from beancount.query import query_env
-from beancount.query import query_parser
+from beanquery import query_compile
+from beanquery.parser.parser import KEYWORDS
```

Comment:
Same for `beanquery.parser`. In particular `beanquery.parser.parser` is generated by TaTsu, thus anything in there that is not generated parser API is subject to change as the TaTsu code generation evolved. Most likely `KEYWORDS` is not going anywhere, but it is not under the `beanquery` developers control.

Reply:
this file `contrib/scripts` generates the JSON file for the frontend query autocompletion, which only needs to run with a fixed version of beanquery (constraints.txt) at irregular points in time, so I'm fine with using internal implementation details here

Reply:
(and constraints.txt fixes Tatsu as well)

Reply:
Sure, `beanquery.parser.parser.KEYWORDS` is fairly safe to use.